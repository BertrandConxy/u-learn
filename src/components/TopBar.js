import React from 'react';
import PropTypes from 'prop-types';
import AppBar from '@mui/material/AppBar';
import Toolbar from '@mui/material/Toolbar';
import IconButton from '@mui/material/IconButton';
import MenuIcon from '@mui/icons-material/Menu';
import Box from '@mui/material/Box';
import ArrowDropDownIcon from '@mui/icons-material/ArrowDropDown';
import { FaUser } from 'react-icons/fa';

const drawerWidth = 300;
const TopBar = ({ handleDrawerToggle }) => (
  <Box sx={{ display: 'flex' }}>
    <AppBar
      color="transparent"
      position="fixed"
      sx={{
        width: { sm: `calc(100% - ${drawerWidth}px)` },
        ml: { sm: `${drawerWidth}px` },
        boxShadow: 'none',
      }}
    >
      <Toolbar>
        <IconButton
          color="inherit"
          aria-label="open drawer"
          edge="start"
          onClick={handleDrawerToggle}
          sx={{ mr: 2, display: { sm: 'none' } }}
        >
          <MenuIcon />
        </IconButton>
        <Box
          sx={{
            display: 'flex',
            alignItems: 'center',
            justifyContent: 'flex-end',
            width: '100%',
            mr: 10,
          }}
        >
          <div className="image-holder">
            <img
              src="https://countryflagsapi.com/png/in"
              alt="logo"
              style={{ width: '100%', borderRadius: '1rem' }}
            />
          </div>
          <ArrowDropDownIcon />
        </Box>
        <Box
          sx={{
            display: 'flex',
            alignItems: 'center',
            justifyContent: 'flex-end',
          }}
        >
          <div className="user-round">
            <FaUser className="icon" />
          </div>
        </Box>
      </Toolbar>
    </AppBar>
  </Box>
);

TopBar.propTypes = {
  handleDrawerToggle: PropTypes.func.isRequired,
};

export default TopBar;
