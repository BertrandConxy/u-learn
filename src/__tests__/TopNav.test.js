import renderer from 'react-test-renderer';
import TopBar from '../components/TopBar';

describe('TopNav', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<TopBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
