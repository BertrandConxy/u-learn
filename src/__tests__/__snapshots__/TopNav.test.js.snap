// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNav renders correctly 1`] = `
<div
  className="MuiBox-root css-k008qs"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionFixed mui-fixed css-e3y1ci-MuiPaper-root-MuiAppBar-root"
    style={
      Object {
        "visibility": undefined,
      }
    }
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-6iz4mj-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
      <div
        className="MuiBox-root css-13q6wg9"
      >
        <div
          className="image-holder"
        >
          <img
            alt="logo"
            src="https://countryflagsapi.com/png/in"
            style={
              Object {
                "borderRadius": "0.6rem",
                "width": "100%",
              }
            }
          />
        </div>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m7 10 5 5 5-5z"
          />
        </svg>
      </div>
      <div
        className="MuiBox-root css-192qrng"
      >
        <a
          className="link"
          href="#login"
        >
          <div
            className="user-round"
          >
            <svg
              className="icon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
              />
            </svg>
          </div>
        </a>
      </div>
    </div>
  </header>
</div>
`;
